### PR TITLE
[EXL-146] docs: 📘 Overall writing enhancement

### PR DESCRIPTION
**Pull request by roy-shiloh**
Overall writing enhancement

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] UI changes have been reviewed
- [x] No UI review needed

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] Other... Please describe:

## What is the new behavior?

Modified docs

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by roy-shiloh** [comment]
@amir-ben-shimol Can you resolve the checks please?

**Comment by roy-shiloh** [comment]
@tal-rofe Please approve

**Review by amir-ben-shimol — APPROVED** [review]

File: .github/workflows/integrate.yaml [hunk 1]
```diff
@@ -19,10 +19,7 @@ jobs:
                   node-version: 16
 
             - name: Installing dependencies
-              run: pnpm install
+              run: npm install
```
Comment (tal-rofe) [inline .github/workflows/integrate.yaml:22]: https://github.com/Exlint/node-environment

Use this action

File: .github/workflows/integrate.yaml [hunk 2]
```diff
@@ -19,10 +19,7 @@ jobs:
...
 
             - name: Check unused dependencies
-              run: pnpm depcheck
-
-            - name: Check Files and Directories Linter
-              run: pnpm inflint
+              run: npm run depcheck
```
Comment (tal-rofe) [inline .github/workflows/integrate.yaml:25]: the repository uses `pnpm`, not `npm`

File: package.json [hunk 3]
```diff
@@ -4,7 +4,6 @@
 	"description": "Exlint Docs",
 	"private": true,
 	"scripts": {
-		"preinstall": "npx only-allow pnpm",
```
Comment (tal-rofe) [inline package.json:7]: revert it back

**Review by tal-rofe — CHANGES_REQUESTED** [review]

File: .github/workflows/integrate.yaml [hunk 4]
```diff
@@ -4,25 +4,77 @@ on:
...
+            runner: ubuntu-latest
+
+    test_depcheck:
+        needs: call_pnpm_caching_workflow
+        name: Testing Depcheck
+
         runs-on: ubuntu-latest
+        steps:
+            - name: Setup Node environment
+              uses: Exlint/node-environment@v1.1.0
```
Comment (tal-rofe) [inline .github/workflows/integrate.yaml:25]: you need to upgrade this action version

File: .github/workflows/integrate.yaml [hunk 5]
```diff
@@ -4,25 +4,77 @@ on:
     pull_request:
         branches: [main]
 
+permissions:
+    contents: read
+
 jobs:
-    test_pull_request:
-        name: Testing & Checking the code
+    call_pnpm_caching_workflow:
```
Comment (tal-rofe) [inline .github/workflows/integrate.yaml:11]: dont use it and delete its file "./.github/workflows/pnpm-caching.yaml"

File: .github/workflows/integrate.yaml [hunk 6]
```diff
@@ -4,25 +4,77 @@ on:
...
+        name: Set pnpm caching
+
+        uses: ./.github/workflows/pnpm-caching.yaml
+        with:
+            runner: ubuntu-latest
+
+    test_depcheck:
```
Comment (tal-rofe) [inline .github/workflows/integrate.yaml:18]: no `test_` prefix (all)

File: .github/workflows/integrate.yaml [hunk 7]
```diff
@@ -4,25 +4,77 @@ on:
...
+        needs: call_pnpm_caching_workflow
+        name: Testing Depcheck
+
         runs-on: ubuntu-latest
+        steps:
+            - name: Setup Node environment
+              uses: Exlint/node-environment@v1.1.0
+              with:
+                  node-version: 18.12.1
```
Comment (tal-rofe) [inline .github/workflows/integrate.yaml:27]: upgrade version

File: .github/workflows/integrate.yaml [hunk 8]
```diff
@@ -4,25 +4,77 @@ on:
...
+                  package-manager: pnpm
+                  package-manager-version: 7.19.0
```
Comment (tal-rofe) [inline .github/workflows/integrate.yaml:29]: upgrade version

**Review by tal-rofe — CHANGES_REQUESTED** [review]

**Review by tal-rofe — APPROVED** [review]

**Pull request merged**